src/olympia/reviewers/serializers.py

```diff
@@ -307,7 +307,7 @@ def get_diff(self, obj):
             parent=parent,
             pathspec=[self.get_selected_file(obj)])
 
-        return diff
+        return next(iter(diff), None)
```

Comment:
I think we should be implementing this in `lib/git::get_diff` instead to make sure we implement this as low-level as possible. Make sure `changes` is a dictionary and render the patch datastructure in there.

Additionally, raise an error in case we hit an edge-case somewhere where it's not the case anymore.

I looked at various diffs and couldn't find a scenario tbh, technically I'm still not 100% sure if there aren't any edgecases I'm not thinking about. But neither for renames, copies or others the diffs aren't issued more than once per file.

Reply:
@EnTeQuAk the `lib/git::get_diff` returns a diff between two commits, which means diffs for different files. Specifying `pathspec` helps focus on one specific file but I believe `pathspec` could also be a partial path like a directory name.

In the serializer, we pass the file name to `lib/git::get_diff` to make sure we retrieve only one diff for this specific file. This is why we can't have more than one diff in the list returned by `lib/git::get_diff`.

I think `lib/git::get_diff` does the right thing and has the correct API/return value. I don't think we should change the return type to be an array, `None` or a single item depending on what's supplied to this function because that'd be consistent.

If we want to implement this in `lib/git`, then I'd rather suggest a `get_diff_for_file` function that calls `get_diff` and returns `next(iter(diff), None)`.

Reply:
You're right, I forgot that this is only because we are specifying `pathspec`.

Maybe add a comment before the `return` mentioning that? Something like, "Because we're always specifying pathspec with the currently selected file we can inline the diff because there will always be one." or so.

With that, I think we should be okay